pkg/apis/pipeline/v1alpha1/task_types.go

```diff
@@ -46,6 +45,8 @@ func (t *Task) Copy() TaskInterface {
 
 // TaskSpec defines the desired state of Task.
 type TaskSpec struct {
+	v1alpha2.TaskSpec `json:",inline"`
```

Comment:
This is not required per-se, it just simplify the conversion using embedded struct *and* it doesn't change on the api side (except user of the go library)

Reply:
If you look at `Pipeline`, it is not using embedding as we can't really do that (same `task` field targeting different type)

Reply:
aha, so this is how TaskSpec.ConvertDown is able [to copy into the v1alpha1.Params and Resources fields](https://github.com/tektoncd/pipeline/pull/2002/files#diff-35489a0872e9373a16f7115dd2618c70R120)?

:+1: